src/varity/hgvs_to_vcf/protein.clj

```diff
@@ -14,11 +16,28 @@
          (keep #(rg/cds->genomic-pos % rg))
          (sort))))
 
-(defn- one-character-substituted?
-  [ref-seq alt-seq pos]
-  (and (not= ref-seq alt-seq)
-       (= (assoc (vec ref-seq) pos nil)
-          (assoc (vec alt-seq) pos nil))))
+(defn- trim-left-right [positions ref alt]
+  (->> (map vector positions ref alt)
+       (drop-while #(apply = (rest %)))
+       reverse
+       (drop-while #(apply = (rest %)))
+       reverse
+       (apply map vector)))
+
+(defn- contiguous? [xs]
+  (when-first [f xs]
+    (let [l (last xs)]
+      (or (= xs (range f (inc l) 1))
+          (= xs (range f (dec l) -1))))))
+
+(defn- codon->variant [positions ref alt]
+  (when-not (= ref alt) ;; not a variant
+    (let [[ps r a] (seq (trim-left-right positions ref alt))]
+      (if (contiguous? ps)
+        [ps (string/join r) (string/join a)]
+        (-> "Ignoring a candidate variant crossing an intron: %s %s %s"
```

Comment:
I feel this message is inappropriate because reason for the ignoring is ambiguous. Such a variant may exist, so `A candidate variant crossing an intron is unsupported` is better.